### PR TITLE
FEATURE: chat redesign - back button to  exit threads

### PR DESCRIPTION
**Pull request by dbattersby**
This is part of the chat redesign work to improve chat navigation.

This change includes:

- New header title **with** channel name (thread list on mobile)
- New header title **without** channel name (thread list on full page chat)
- Removes the close button on threads (mobile only)
- Updates to back button route within thread (mobile), taking user to:
    - The thread index, if they accessed the thread from the thread index.
    - The channel itself, if they accessed the thread directly from the channel.
    - The channel itself, if they accessed the thread from a notification.
- Show thread title in chat drawer header
- Properly convert emoji in thread titles in chat header (all devices)
- Upgrades various templates to use `gjs` format.

/t/-/112524

### CODE REVIEW

File: plugins/chat/assets/javascripts/discourse/components/chat-drawer/header/title.gjs
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/components/chat-drawer/header/title.gjs:26]: why this empty div?

File: plugins/chat/assets/javascripts/discourse/components/chat-drawer/header/title.gjs
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/components/chat-drawer/header/title.gjs:31]: generally I would prefer to avoid any text without any direct wrapper, here you have

div
  icon span
  text
  - text
  
I would recommend having two last text wrapped inside a span

File: plugins/chat/assets/javascripts/discourse/components/chat/thread-list/header.gjs
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/components/chat/thread-list/header.gjs:45]: would need a class

File: plugins/chat/assets/javascripts/discourse/components/chat/thread/header.gjs
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/components/chat/thread/header.gjs:112]: can you change this to use try/catch async/await please?

Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/components/chat/thread/header.gjs:112]: or if you want to limit scope just use "return"

File: plugins/chat/assets/stylesheets/common/chat-thread-list-header.scss
Comment (jjaffeux) [inline plugins/chat/assets/stylesheets/common/chat-thread-list-header.scss:20]: generally speaking styling an html element directly will cause issues

**Comment by jjaffeux** [comment]
@dbattersby I fixed various linting and js errors, I suspect your editor is misconfigured for gjs, let me know if you need help.

Other than this, It's nice if you can fix the few remaining comments, but already ok to merge like this.

**Review by jjaffeux — APPROVED** [review]

**Comment by chancancode** [comment]
👋🏼 just noticed this pull request! I opened #24260 today, but because the process is mostly automated, I could put that on hold and re-run it again after this has landed, if that makes things easier in. In the meantime, if you noticed anything that doesn't seem right in the conversation, feel free to let me know on there.

**Comment by jjaffeux** [comment]
> 👋🏼 just noticed this pull request! I opened #24260 today, but because the process is mostly automated, I could put that on hold and re-run it again after this has landed, if that makes things easier in. In the meantime, if you noticed anything that doesn't seem right in the conversation, feel free to let me know on there.

Thanks, david warned me about the migration you are doing. We will merge this very soon.

**Pull request merged**
